src/Symfony/Component/Security/Ldap/Ldap.php

```diff
@@ -0,0 +1,221 @@
+<?php
+
+namespace Symfony\Component\Security\Ldap;
+
+use Symfony\Component\Security\Ldap\Exception\ConnectionException;
+use Symfony\Component\Security\Ldap\Exception\LdapException;
+
+/**
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ * @author Francis Besset <francis.besset@gmail.com>
+ * @author Charles Sarrazin <charles@sarraz.in>
+ */
+class Ldap implements LdapInterface
+{
+    private $host;
+    private $port;
+    private $version;
+    private $useSsl;
+    private $useStartTls;
+    private $optReferrals;
+    private $connection;
+    private $charmaps;
+
+    /**
+     * Constructor.
+     *
+     * @param string $host
+     * @param int    $port
+     * @param string $dn
+     * @param int    $version
+     * @param bool   $useSsl
+     * @param bool   $useStartTls
+     * @param bool   $optReferrals
+     */
+    public function __construct($host = null, $port = 389, $version = 3, $useSsl = false, $useStartTls = false, $optReferrals = false)
+    {
+        if (!extension_loaded('ldap')) {
+            throw new LdapException('The ldap module is needed.');
+        }
+
+        $this->host = $host;
+        $this->port = $port;
+        $this->version = $version;
+        $this->useSsl = (bool) $useSsl;
+        $this->useStartTls = (bool) $useStartTls;
+        $this->optReferrals = (bool) $optReferrals;
+    }
+
+    public function __destruct()
+    {
+        $this->disconnect();
```

Comment:
Not sure. The LDAP extension actually recommends to use `ldap_close()` at the end of the treatment. I'm not sure it's required, so I didn't remove this code from @lyrixx's PR. See http://php.net/manual/en/ldap.using.php